src/common/ceph_json.h

```diff
@@ -96,28 +259,37 @@ inline std::ostream& operator<<(std::ostream &out, const JSONObj::data_val& dv)
    return out;
 }
 
-class JSONParser : public JSONObj
+class JSONParser final : public JSONObj
 {
-  int buf_len;
+  int buf_len = 0;
   std::string json_buffer;
-  bool success;
+
 public:
-  JSONParser();
-  ~JSONParser() override;
-  void handle_data(const char *s, int len);
+  ~JSONParser() override = default;
 
-  bool parse(const char *buf_, int len);
-  bool parse(int len);
+public:
+  // operate on the internal buffer:
   bool parse();
+  bool parse(int len);
+
+  // operate on a string/stringlike range or object:
+  bool parse(std::string_view sv);
+  bool parse(const std::string& s) { return parse(std::string_view { s }); }
```

Comment:
if you omit this overload, won't `parse("abc"s)` still call the string_view overload?

Reply:
`std::string` supports implicit conversion to `std::string_view` via this conversion operator: https://en.cppreference.com/w/cpp/string/basic_string/operator_basic_string_view

Reply:
I'm not sure why I had that in there, I probably had written it the other way around first for some reason. Good catch, thank you!